### PR TITLE
nonce has to be a uint64 :<

### PR DESCRIPTION
**Pull request by GusCaplan**

### CODE REVIEW

File: src/client/rest/RESTMethods.js [hunk 1]
```diff
@@ -60,6 +60,12 @@ class RESTMethods {
     return new Promise((resolve, reject) => { // eslint-disable-line complexity
       if (typeof content !== 'undefined') content = this.client.resolver.resolveString(content);
 
+      // nonce has to be a uint64 :<
+      if (nonce !== undefined) {
+        nonce = Math.abs(parseInt(nonce));
+        if (isNaN(nonce)) reject(new Error('`nonce` must be a positive integer.'));
```
Comment (GusCaplan) [inline src/client/rest/RESTMethods.js]: consistant return is the bane of my existance

Comment (hydrabolt) [inline src/client/rest/RESTMethods.js]: You could `throw` the error at this stage instead, should be good then 👍

File: src/client/rest/RESTMethods.js [hunk 2]
```diff
@@ -60,6 +60,12 @@ class RESTMethods {
     return new Promise((resolve, reject) => { // eslint-disable-line complexity
       if (typeof content !== 'undefined') content = this.client.resolver.resolveString(content);
 
+      // nonce has to be a uint64 :<
+      if (nonce !== undefined) {
```
Comment (Gawdl3y) [inline src/client/rest/RESTMethods.js]: Use `typeof nonce !== 'undefined'`. It's done literally the line before this.

Comment (Gawdl3y) [inline src/client/rest/RESTMethods.js]: Use `typeof nonce !== 'undefined'`. It's done literally the line before this.

File: src/client/rest/RESTMethods.js [hunk 3]
```diff
@@ -60,6 +60,12 @@ class RESTMethods {
...
+        nonce = Math.abs(parseInt(nonce));
+        if (isNaN(nonce)) throw new Error('`nonce` must be a positive integer.');
```
Comment (Gawdl3y) [inline src/client/rest/RESTMethods.js]: Can't use markdown in error messages.

File: src/client/rest/RESTMethods.js [hunk 4]
```diff
@@ -60,6 +60,12 @@ class RESTMethods {
     return new Promise((resolve, reject) => { // eslint-disable-line complexity
       if (typeof content !== 'undefined') content = this.client.resolver.resolveString(content);
 
+      // nonce has to be a uint64 :<
+      if (nonce !== undefined) {
+        nonce = Math.abs(parseInt(nonce));
```
Comment (Gawdl3y) [inline src/client/rest/RESTMethods.js]: Why? Just add to the validation to ensure it's positive - don't coerce negatives into positives.

**Comment by Gawdl3y** [comment]
Still shouldn't be making negative nonces positive; just throw an error or something.

**Pull request merged**